pkg/kubelet/kubelet.go

```diff
@@ -1130,10 +1130,23 @@ func (kl *Kubelet) syncPod(pod *api.Pod, mirrorPod *api.Pod, runningPod kubecont
 	}
 	kl.volumeManager.SetVolumes(pod.UID, podVolumes)
 
-	podStatus, err := kl.generatePodStatus(pod)
-	if err != nil {
-		glog.Errorf("Unable to get status for pod %q (uid %q): %v", podFullName, uid, err)
-		return err
+	// If we end up here with a create event for an already running pod, it could result in a
+	// restart of its containers. This cannot happen because the delete before the second
+	// create is processed by SyncPods, which cancels this pod worker.
```

Comment:
This seems quite volatile... Maybe validate it by checking if cache knows anything about given Pod in case someone will be playing with this workflow? Or at least add the comment about this assumption (i.e. that we don't call create on existing pods) before this function.